plan/cbo_test.go

```diff
@@ -451,6 +459,19 @@ func (s *testAnalyzeSuite) TestAnalyze(c *C) {
 			sql:  "analyze table t2 index",
 			best: "Analyze{Index(a),Index(b)}",
 		},
+		// Test partitioned table.
+		{
+			sql:  "select * from t4 where t4.a <= 2",
+			best: "UnionAll{TableReader(Table(t4)->Sel([le(test.t4.a, 2)]))->TableReader(Table(t4)->Sel([le(test.t4.a, 2)]))}",
```

Comment:
seems like partition pruning is not working

Reply:
Partition pruning works right, `t4.a <= 2` will generate the `union all `.
I have thought you mean `Table(t4)->Sel([le(test.t4.a, 2)])` should be `IndexLookup`

Reply:
The plan is correct, it should be table scan.